src/internal_events/aws_ecs_metrics.rs

```diff
@@ -43,53 +23,91 @@ impl InternalEvent for AwsEcsMetricsRequestCompleted {
 #[derive(Debug)]
 pub struct AwsEcsMetricsParseError<'a> {
     pub error: serde_json::Error,
-    pub url: &'a str,
+    pub endpoint: &'a str,
     pub body: Cow<'a, str>,
 }
 
 impl<'a> InternalEvent for AwsEcsMetricsParseError<'_> {
     fn emit_logs(&self) {
-        error!(message = "Parsing error.", url = %self.url, error = %self.error);
+        error!(
+            message = "Parsing error.",
+            endpoint = %self.endpoint,
+            error = ?self.error,
+            stage = "processing",
+            error_type = "parse_failed",
+        );
         debug!(
             message = %format!("Failed to parse response:\\n\\n{}\\n\\n", self.body.escape_debug()),
-            url = %self.url,
+            endpoint = %self.endpoint,
             internal_log_rate_secs = 10
         );
     }
 
     fn emit_metrics(&self) {
         counter!("parse_errors_total", 1);
+        counter!(
+            "component_errors_total", 1,
+            "stage" => "processing",
+            "error_type" => "parse_failed",
+            "endpoint" => self.endpoint.to_owned(),
+        );
     }
 }
 
 #[derive(Debug)]
-pub struct AwsEcsMetricsErrorResponse<'a> {
+pub struct AwsEcsMetricsResponseError<'a> {
     pub code: hyper::StatusCode,
-    pub url: &'a str,
+    pub endpoint: &'a str,
 }
 
-impl InternalEvent for AwsEcsMetricsErrorResponse<'_> {
+impl InternalEvent for AwsEcsMetricsResponseError<'_> {
     fn emit_logs(&self) {
-        error!(message = "HTTP error response.", url = %self.url, code = %self.code);
+        error!(
+            message = "HTTP error response.",
+            endpoint = %self.endpoint,
+            code = %self.code,
```

Comment:
Should the log and metric label's for `code` be renamed `error` for consistency?